utreexo-p2p-bip.md

```diff
@@ -1,236 +1,426 @@
-In order start the 1600 through 2400 IBD task, however, the node should know what the state of the utxo set is at block 1600, so that it can validate and modify the accumulator.  In order to do this, the binary can provide "linkup hints", where the state of the accumulator is given for a desired block hash.  While giving the state of the system might seem at first glance to be introducing a trust assumption, these are not trusted states; the node performing IBD tries out the state given for a block height, but checks that when that state is reached from the thread "below" that it properly links up, with the accumulator state arrived at through full validation matching the state given.  If that link up does not successfully happen, the IBD process should halt; these hints are statements of fact that are hard-coded into the program itself, and if they are false all bets are off about the program.
+### Utreexo TTL
 
-The format of the Linkup hints is:
+| Field        | Type                | Description                                                                                                                                                          |
+|--------------|---------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| block height | uint32              | The time-to-live value of a leaf in the Utreexo merkle forest. The value is determined by the amount of leaves that were added to the accumulator since its creation |
+| length       | varint              | The length of the TTLs                                                                                                                                               |
+| TTLs         | vector of TTL infos | position in the Utreexo merkle forest when the leaf was removed                                                                                                      |
 
-Blockhash, Numleaves, [Roots]
+## New Messages
 
-Archive nodes create a forest of Linkup hints, so that they can prove, with respect to the Linkup forest roots in a node performing IBD, what their binary has claimed the utxo accumulator state to be at any block height.
+### MSG_UTREEXO_PROOF
 
-# Proof overshoot
+`MSG_UTREEXO_PROOF` is all the data required for a CSN or archive node using the Utreexo accumulators to validate a Bitcoin block.
 
-Archive nodes continually update their Linkup and Block Summary forests, and provide up to date proofs from both.  Nodes which have a committed root set in the past may appear to have an incompatible root set due to a different set of roots than the ones the archive node is using to prove.  For the node performing IBD, the proof prefix sent by the archive node will be compatible.
+Its [BIP324 P2PV2](https://github.com/bitcoin/bips/blob/master/bip-0324.mediawiki#user-content-v2_Bitcoin_P2P_message_structure) message type is `29`.
```

Comment:
(The thing I've mentioned over signal)

Apart from the BIP324 message id we need the V1 cmdString, I think for this one is `uproof` right?

Reply:
Ohh right I get what you mean now. I'll add those.

They're just not called cmdString in btcd code heh